src/main/java/mekanism/client/render/hud/MekaSuitHUD.java

```diff
@@ -31,28 +34,46 @@ public void render(ForgeIngameGui gui, PoseStack poseStack, float partialTicks,
         Minecraft minecraft = Minecraft.getInstance();
         if (!minecraft.options.hideGui && !minecraft.player.isSpectator() && MekanismConfig.client.enableHUD.get() && MekanismClient.renderHUD) {
             int count = 0;
-            Map<EquipmentSlot, List<Component>> renderStrings = new LinkedHashMap<>();
+            final List<List<Component>> renderStrings = new LinkedList<>();
             for (EquipmentSlot slotType : EQUIPMENT_ORDER) {
                 ItemStack stack = minecraft.player.getItemBySlot(slotType);
                 if (stack.getItem() instanceof IItemHUDProvider hudProvider) {
                     List<Component> list = new ArrayList<>();
                     hudProvider.addHUDStrings(list, minecraft.player, stack, slotType);
                     int size = list.size();
                     if (size > 0) {
-                        renderStrings.put(slotType, list);
+                        renderStrings.add(list);
                         count += size;
                     }
                 }
             }
+            if (Mekanism.hooks.CuriosLoaded) {
+                final var invOptional = CuriosIntegration.getCuriosInventory(minecraft.player);
+                if (invOptional.isPresent()) { // Can't use ifPresent due to count not being final
+                    final IItemHandler inv = invOptional.get();
+                    for (int i = 0; i < inv.getSlots(); i++) {
+                        ItemStack stack = inv.getStackInSlot(i);
+                        if (stack.getItem() instanceof IItemHUDProvider hudProvider) {
+                            List<Component> list = new ArrayList<>();
+                            hudProvider.addCurioHUDStrings(list, minecraft.player, stack);
+                            int size = list.size();
+                            if (size > 0) {
+                                renderStrings.add(list);
+                                count += size;
+                            }
+                        }
```

Comment:
This could just be a helper method that returns the size to increment count by and then reduces the duplicate code between this and the normal armor spots.